subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -665,6 +668,12 @@ This is something you can combine with <<continuous_build, continuous build>> us
 gradle test --continuous --tests "com.mypackage.foo.*"
 ----
 
+[NOTE]
+====
+Since `--tests` supports method name matching, Gradle has to load all possible classes, which may lead to very bad performance, especially with api:org.gradle.api.tasks.testing.Test#getForkEvery()[`forkEvery`].
+To help Gradle exclude as many classes as possible, please don't use a pattern starting with a wildcard (*).
```

Comment:
❌ If we don't want people to use a wildcard at the front, why do we show exactly that in the examples above? I think we just shouldn't show this if it's bad for performance. With your improvements it really shouldn't be necessary anymore to use a leading * anywhere.